packages/stream/common/types.ts

```diff
@@ -171,3 +196,21 @@ export interface Stream {
 
   unlocked(currentTimestamp: number): BN;
 }
+
+export class ContractError extends Error {
+  public contractErrorCode: string | null;
+
+  constructor(error: Error, code?: string | null) {
+    super(error.message); // Call the base class constructor with the error message
```

Comment:
Nit: make these comments as jsdoc.
https://jsdoc.app/about-getting-started.html

This way they will be seen in the IDE